app/views/gardens/_actions.html.haml

```diff
@@ -21,3 +21,8 @@
               class: 'btn btn-default', id: 'delete_garden_link' do
       %span.glyphicon.glyphicon-trash{ title: "Delete" }
       Delete
+  - if can?(:edit, @garden) && can?(:create, Photo)
+    = link_to new_photo_path(type: "garden", id: @garden.id),
+    class: 'btn btn-primary' do
```

Comment:
The `class` here should be indented level with `new_photo_path`. I'm not actually sure this will work - HAML is whitespace-sensitive. Could you add a unit test for this behaviour? Ignore any "don't add new view specs" comments you find, we've been meaning to delete those.